crux_core/src/capability.rs

```diff
@@ -266,37 +266,92 @@ where
 /// }
 /// ```
 ///
-pub struct CapabilityContext<Ef, Event> {
-    inner: std::sync::Arc<ContextInner<Ef, Event>>,
+pub struct CapabilityContext<Op, Event>
+where
+    Op: Operation,
+{
+    inner: std::sync::Arc<ContextInner<Op, Event>>,
+}
+
+struct ContextInner<Op, Event>
+where
+    Op: Operation,
+{
+    shell_channel: Sender<Step<Op>>,
+    app_channel: Sender<Event>,
+    spawner: crate::executor::Spawner,
 }
 
-struct ContextInner<Ef, Event> {
-    steps: Sender<Step<Ef>>,
-    events: Sender<Event>,
+pub struct ProtoContext<E, Event> {
```

Comment:
Small thing, but `E` is quite ambiguous (particularly when there's already `Event`).  `Eff`, or `Effect` maybe?